vsintegration/src/FSharp.Editor/Refactor/AddExplicitReturnType.fs

```diff
@@ -0,0 +1,125 @@
+﻿namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System.Composition
+open System.Threading
+open System.Threading.Tasks
+open FSharp.Compiler.CodeAnalysis
+open FSharp.Compiler.Symbols
+open FSharp.Compiler.Text
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeRefactorings
+open Microsoft.CodeAnalysis.CodeActions
+open CancellableTasks
+
+[<ExportCodeRefactoringProvider(FSharpConstants.FSharpLanguageName, Name = "AddExplicitReturnType"); Shared>]
+type internal AddExplicitReturnType [<ImportingConstructor>] () =
+    inherit CodeRefactoringProvider()
+
+    static member isValidMethodWithoutTypeAnnotation
+        (symbolUse: FSharpSymbolUse)
+        (parseFileResults: FSharpParseFileResults)
+        (funcOrValue: FSharpMemberOrFunctionOrValue)
+        =
+        let typeAnnotationRange =
+            parseFileResults.TryRangeOfReturnTypeHint(symbolUse.Range.Start, false)
```

Comment:
Ooooph probably worth renaming now, since it is not used for hints anymore :)

Reply:
It is used to replace the return type hint with an explicit return type. For that I get the range where the hint virtually is.  If this returns None, then there is already an explicit type present

Reply:
No sure, I just don't like the Hint notion here since we are in a different context. I'd say maybe `TryRangeOfReturnType` then.

Reply:
It is exactly that of the different context XD 
![image](https://github.com/dotnet/fsharp/assets/19798013/c3388d45-0a4f-473d-9fc0-8a5c8baca80b)
The red marker are Hints, right?

Reply:
Right, yeah okay, it's probably appropriate here, sorry for the hiccup then, let's keep the name :)

Reply:
No Problem :)